src/Service/OrderCartAssociationService.php

```diff
@@ -57,7 +58,7 @@ public function __construct(CartDuplicationService $cartDuplication)
 	public function createPendingCart(Order $order)
 	{
 		// globally restores the cart.
-		$newCartId = $this->cartDuplication->restoreCart($order->id_cart);
+		$newCartId = $this->cartDuplication->restoreCart($order->id_cart, Config::RESTORE_CART_BACKTRACE_MEMORIZATION_SERVICE);
```

Comment:
first of all I would prefer **enum** usage in PHP https://stackoverflow.com/questions/254514/php-and-enumerations . Something like this or this package https://packagist.org/packages/myclabs/php-enum

what it gives is type strictness so we cant pass any random string - we would receive error. 

Also what else it gives is that you can warp related configuration and document it as well.

Reply:
This is aswell now optional suggestion, but if you have a bit spare time I would dive into this implementation and would deprecate remaining configs